classes/controller/ProductListingFrontController.php

```diff
@@ -497,24 +500,48 @@ protected function getAjaxProductSearchVariables()
     /**
      * Cleans the products array with only whitelisted properties.
      *
-     * @return array
+     * @param array[] $products
+     *
+     * @return array[] Filtered product list
      */
     protected function prepareProductArrayForAjaxReturn(array $products)
     {
-        $allowed_properties = array('id_product', 'price', 'reference', 'active', 'description_short', 'link',
-            'link_rewrite', 'name', 'manufacturer_name', 'position', 'cover', 'url', 'canonical_url', 'add_to_cart_url',
-            'has_discount', 'discount_type', 'discount_percentage', 'discount_percentage_absolute', 'discount_amount',
-            'price_amount', 'regular_price_amount', 'regular_price', 'discount_to_display', 'labels', 'main_variants',
-            'unit_price', 'tax_name', 'rate'
-        );
-        foreach ($products as $product_key => $product) {
-            foreach ($product as $product_property => $data) {
-                if (!in_array($product_property, $allowed_properties)) {
-                    unset($products[$product_key][$product_property]);
-                }
-            }
-        }
-        return $products;
+        $filter = (new CollectionFilter())
```

Comment:
you have access to service container in controllers, take a look at config/services/services/front folder :)

Reply:
Good to know! But how would modules override that?

Reply:
You have information in docs you asked me to do :) 

tldr: write a yml file and a class which implement the same interface in the module

Reply:
a real exemple here: https://github.com/PrestaShop/prestafony-project/blob/master/modules/ps_test/config/services.yml

Reply:
Ohh, that could be the answer to the extensibility issue I'm struggling with!

Reply:
<3 <3 Always happy to help <3<3